charts/kube-state-metrics/templates/service.yaml

```diff
@@ -28,6 +28,12 @@ spec:
     nodePort: {{ .Values.service.nodePort }}
   {{- end }}
     targetPort: 8080
+  {{ if .Values.selfMonitor.enabled }}
+  - name: "metrics"
+    protocol: TCP
+    port: {{ .Values.selfMonitor.telemetryPort | default 8081 }}
+    targetPort: {{ .Values.selfMonitor.telemetryPort | default 8081 }}
```

Comment:
can you hardwire the targetPort as done in the service spec above?

Reply:
You mean setting it 8081 in any case?
I added that check because could be changed the telemetry port where kube-state-metrics listen with `--telemetry-port` I included in the deployment
Sorry if I do not understand you correctly.

Reply:
Yes, I'd suggest to hardcode both of the occurences and set them to the default. Users then have the choice from outside on which port to make it visible.

Reply:
in that case, it would be necessary to remove the `telemetry-port` from the deployment to disable to possibility of changing where kube-state-metrics listens for telemetry, right?

Reply:
Yes, or hardcode it to 8081 if you want to be explicit there.

Reply:
Ok, understood. Done